Crash Report 3.1.2 Custom Apps Parameter specified as non-null is null

**Describe the bug**
Android reports a directory as null

**Expected behavior**
Safeguard against this error with nullable types

**Environment**
Device:[on7xelte]
Model:[SM-G610F]
Manufacturer:[samsung]
Time:[1567686317000]
Android Version:[8.1.0]

**Logs**
```
Caused by: java.lang.IllegalArgumentException: Parameter specified as non-null is null: method io.reactivex.plugins.RxJavaPlugins.walk, parameter $this$walk
at io.reactivex.plugins.RxJavaPlugins.walk(RxJavaPlugins.java:2)
at org.kiwix.kiwixmobile.custom.main.CustomFileValidator.scanDirs(CustomFileValidator.kt:3)
at org.kiwix.kiwixmobile.custom.main.CustomMainActivity.openObbOrZim(CustomMainActivity.kt:2)
at org.kiwix.kiwixmobile.custom.main.CustomMainActivity.onCreate(CustomMainActivity.kt:7)
```

--- a/custom/src/main/java/org/kiwix/kiwixmobile/custom/main/CustomFileValidator.kt
+++ b/custom/src/main/java/org/kiwix/kiwixmobile/custom/main/CustomFileValidator.kt
@@ -52,10 +52,10 @@ class CustomFileValidator @Inject constructor(private val context: Context) {
   private fun zimFiles() =
     scanDirs(ContextCompat.getExternalFilesDirs(context, null), "zim")
 
-  private fun scanDirs(dirs: Array<out File>, extensionToMatch: String): List<File> =
-    dirs.fold(listOf()) { acc, dir ->
+  private fun scanDirs(dirs: Array<out File?>?, extensionToMatch: String): List<File> =
+    dirs?.filterNotNull()?.fold(listOf()) { acc, dir ->
       acc + dir.walk().filter { it.extension.startsWith(extensionToMatch) }.toList()
-    }
+    } ?: emptyList()
 }
 
 sealed class ValidationState {

